src/ragas/metrics/context_precision.py

```diff
@@ -177,9 +186,75 @@ def _score_batch(
 
 
 @dataclass
-class ContextRelevancy(ContextPrecision):
-    name: str = "context_relevancy"
-    show_deprecation_warning: bool = True
+class ContextPrecision(MetricWithLLM):
+
+    """
+    Average Precision is a metric that evaluates whether all of the
+    relevant items selected by the model are ranked higher or not.
+
+    Attributes
+    ----------
+    name : str
+    batch_size : int
+        Batch size for openai completion.
+    """
+
+    name: str = "context_precision"
+    evaluation_mode: EvaluationMode = EvaluationMode.qc
+    batch_size: int = 15
+
+    def init_model(self: t.Self):
+        pass
+
```

Comment:
this is not needed because the base class for `MetricsWithLLM` implements this now. this will override that implementation